src/components/generics/date-range-picker.tsx

```diff
@@ -0,0 +1,90 @@
+import { Box } from "@mui/material";
+import { DatePicker } from "@mui/x-date-pickers";
+import { DateTime } from "luxon";
+import { useEffect, useState } from "react";
+import { getTimeDifferenceInDays } from "../../utils/time-utils";
+
+/**
+ * Component properties
+ */
+interface DateRangePickerProps {
+  dateTimeNow: DateTime;
+  setDates: Function;
```

Comment:
What does this setDates: Function mean here?

Reply:
It means that the type of setDates prop is "function". 

More elaborate typing would be something like this:
```suggestion
  interface DateRangePickerProps {
  dateTimeNow: DateTime;
  setDates: (startDate: DateTime | null | undefined, endDate: DateTime | null | undefined, days : number | undefined) => void
```
But my quick testing of this created problems (is possibly undefined, cannot be assigned to type undefined | null) around the code, which would be fixed by validating the code with if-checks or ternary operators, or maybe using the hasAllPropsDefined()-function.

Reply:
I managed to make the changes Petri suggested and solve the problems with it.